test/unit/utils/FactoryManager.mocha.ts

```diff
@@ -0,0 +1,121 @@
+/// <reference path='../test.mocha.ts' />
+
+// A simple factory stub
+class FactoryStub {
+
+  public args:any[];
+
+  constructor(...args:any[]){
+    this.args = args;
+  }
+}
+
+describe('n3Charts.Utils.FactoryManager', () => {
+  var factoryMgr: n3Charts.Utils.FactoryManager = undefined;
+
+  beforeEach(() => {
+
+    factoryMgr = new n3Charts.Utils.FactoryManager();
+  });
+
+  describe('index()', () => {
+
+    it('should return the index of the factory entry', () => {
+
+      var f = factoryMgr
+        .register('stub1', FactoryStub)
```

Comment:
I think `register` should return the newly registered factory, and we should have a method called `registerMany` or something... Thoughts ?